ci.sh

```diff
@@ -1,8 +1,36 @@
 #!/usr/bin/env bash
 
-set -e
+set -eu
 set -o pipefail
 
+function install_node() {
+    # if an existing nvm is already installed we need to unload it
+    nvm unload 2> /dev/null || true
+    rm -rf ./__nvm/ && git clone --depth 1 https://github.com/creationix/nvm.git ./__nvm
+    # note: we must temporarily disable throwing on unbound
+    # variables because nvm has them
+    set +eu
+    source ./__nvm/nvm.sh
+    nvm install ${1}
+    nvm use ${1}
+    set -eu
+    node --version
+    npm --version
+    which node
+    which npm
+    set +eu
+    set +o pipefail
+}
+
+
+install_node 4
+
+if [[ "$(git symbolic-ref --short -q HEAD)" == "master" ]]; then
+    npm update
+else
+    npm install
+fi
```

Comment:
Are there any consequences of moving this from the `dependencies` step to the `test` step?  Or is that just presentational?